yb-voyager/cmd/importData.go

```diff
@@ -798,6 +804,71 @@ func importData(importFileTasks []*ImportFileTask, errorPolicy importdata.ErrorP
 	fmt.Printf("\nImport data complete.\n")
 }
 
+// For a fresh start but non empty tables in tableList && OnPrimaryKeyConflict is set to IGNORE -> notify user
+func runPKConflictModeGuardrails(importFileTasks, pendingTasks, completedTasks []*ImportFileTask) error {
+	// in case of ERROR mode, no need to check for non-empty tables
+	// but for IGNORE or UPDATE(in future), we need to prompt user
+	if tconf.OnPrimaryKeyConflictAction == constants.PRIMARY_KEY_CONFLICT_ACTION_ERROR {
+		return nil
+	}
+
+	if !isTargetDBImporter(importerRole) {
+		return nil
+	}
+
+	if !isFreshStart(importFileTasks, pendingTasks, completedTasks) {
```

Comment:
(for discussion, not blocker)
I'm wondering if we really need this check. 
Can we blindly just check for all pendingTasks every time? 

the good: 
- this will help with resumption cases as well. I.e. if you stop in the middle, and then resume, and there are some tables for which we have not yet imported, this will check and prompt for those tables as well. 
the bad: 
- in import-data-file case, there can be multiple tasks for the same table, so we will unnecessarily prompt in those cases